api/internal/localizer/util.go

```diff
@@ -132,25 +135,65 @@ func locFilePath(fileURL string) string {
 	// File urls must have http or https scheme, so it is safe to use url.Parse.
 	u, err := url.Parse(fileURL)
 	if err != nil {
-		log.Fatalf("cannot parse validated file url %q: %s", fileURL, err.Error())
+		log.Panicf("cannot parse validated file url %q: %s", fileURL, err)
 	}
 
-	// Percent-encodings should be preserved in case sub-delims have special meaning.
+	// We include both userinfo and port, lest they determine the file read.
+	authority := strings.TrimPrefix((&url.URL{
+		User: u.User,
+		Host: u.Host,
+	}).String(), "//")
```

Comment:
I would really prefer not to include the `port` or `userinfo` in the directory structure. From your PR description, it sounds like you can easily remove them if you need to. I think it adds unnecessary noise and complexity to the resulting directory structure. I also very much doubt that the end-user will want their user-info to end up being part of something they push to a shared repository. We can consider adding it later if we see demand for it.